core/src/main/java/org/elasticsearch/bootstrap/JNACLibrary.java

```diff
@@ -39,6 +39,13 @@
     public static final int MCL_CURRENT = 1;
     public static final int ENOMEM = 12;
     public static final int RLIMIT_MEMLOCK = Constants.MAC_OS_X ? 6 : 8;
+
+    // this is only valid on Linux and the value *is* different on OS X
+    // see /usr/include/sys/resource.h on OS X
+    // on Linux the resource RLIMIT_NPROC means *the number of threads*
+    // this is in opposition to BSD-derived OSes
+    public static final int RLIMIT_NPROC = 6;
+
```

Comment:
the one thing that sucks about it being here is it kinda violates the organization we have: where anything in this JNACLibrary is posix stuff.

That is one reason RLIMIT_NPROC is also defined in Seccomp.java, but otherwise it is using this JNAClibrary for the portable stuff. In that file, its being used for the BSD case, and note that constant has a different value.

So it keeps some organization, that JNACLibrary is portable stuff, and the non-portable stuff goes somewhere else... thats why the constant is defined there and not here. For the same reason it seems odd to have the Linux constant.